services/matrix/matrix.service.js

```diff
@@ -59,27 +85,55 @@ module.exports = class Matrix extends BaseJsonService {
       errorMessages: {
         401: 'auth failed',
         403: 'guests not allowed',
-        429: 'rate limited by rooms host',
+        429: 'rate limited by remote server',
       },
     })
   }
 
-  async fetch({ host, roomId }) {
-    let auth
-    try {
-      auth = await this.registerAccount({ host, guest: true })
-    } catch (e) {
-      if (e.prettyMessage === 'guests not allowed') {
-        // attempt fallback method
-        auth = await this.registerAccount({ host, guest: false })
-      } else throw e
+  async lookupRoomAlias({ host, roomAlias, accessToken }) {
+    return this._requestJson({
+      url: `https://${host}/_matrix/client/r0/directory/room/%23${roomAlias}`,
```

Comment:
What characters are valid for a room alias? Would it be better to use 
```
`https://${host}/_matrix/client/r0/directory/room/${encodeURIComponent(`#${roomAlias}`)}`
```

in case there is some other character that needs URL-encoding?

Reply:
It should be fairly okay without it, but better safe than sorry. Good catch.

Reply:
Done in https://github.com/badges/shields/pull/2527/commits/2a94b15e0c5f92bcb17333e187087d3accf9c6e1